### DISCUSSION TITLE
blog/angular-private-services

### DISCUSSION THREAD

**Discussion by yjaaidi** [marmicode/marmicode#7] [General]
Let's talk about https://marmicode.io/blog/angular-private-services

**Comment by Lonli-Lokli** [comment]
Will it be possible to use same singleton service in multiple places, but without parent component or ngmodule?

It's quite interesting topic of shared service between two standalone components

**Reply by yjaaidi** [reply]
Interesting question @Lonli-Lokli.
What would be the use case? 
You can already do that using `{providedIn: 'root'}` but in case you want to share multiple instances between different types of components, this might come in handy: https://stackblitz.com/edit/configurable-shared-services?file=src%2Fapp%2Fapp.component.ts 

The only use case I can think about right now is providing a library where you want your components to communicate together without anyone being able to hack into that communication channel.

**Reply by Lonli-Lokli** [reply]
So obvios usecase is to share data aka rxjs store between components, eg CounterService to be the same. Your example with root is good but this services will live foreever right? And not destroyed with component
